### PR TITLE
feat(back): #1218 add python 3.12 support

### PR DESCRIPTION
**Pull request by rohaquinlop**
- add python 3.12 support
- update all builtins to use python 3.12

### CODE REVIEW

File: makes/utils/makePythonLock/update.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 function main {
   local cmd=(just m . /utils/makePythonLock)
 
-  : \
+  : \  \
```
Comment (dsalaza4) [inline makes/utils/makePythonLock/update.sh:6]: Typo?

Comment (rohaquinlop) [reply makes/utils/makePythonLock/update.sh:6]: No, It was added by `formatBash`

Comment (dsalaza4) [reply makes/utils/makePythonLock/update.sh:6]: Maybe making  the first command look like `: && other \` would make this look better 🤔

Comment (rohaquinlop) [reply makes/utils/makePythonLock/update.sh]: On it!

**Review by dsalaza4 — APPROVED** [review]
lgtm

**Pull request merged**